wolvic/browser/session_settings.cc

```diff
@@ -39,8 +44,45 @@ SessionSettings::UserAgentMode SessionSettings::GetUserAgentMode() const {
 }
 
 void SessionSettings::SetUserAgentOverride(
-    const absl::optional<std::string>& value) {
-  user_agent_override_ = value;
+    const absl::optional<std::string>& ua_string_override) {
+  user_agent_override_ = ua_string_override;
+  if (!ua_string_override || !web_contents()) {
+    return;
+  }
+
+  blink::UserAgentOverride override_ua_with_metadata;
+  override_ua_with_metadata.ua_string_override = *ua_string_override;
+
+  // If kUACHOverrideBlank is enabled, set user-agent metadata with the
+  // default blank value.
+  if (ua_string_override && !ua_string_override->empty() &&
+      base::FeatureList::IsEnabled(blink::features::kUACHOverrideBlank)) {
+    override_ua_with_metadata.ua_metadata_override =
+      blink::UserAgentMetadata();
+  }
+
+  // Generate user-agent client hints in the following three cases:
+  // 1. If user provide the user-agent metadata overrides, we use the
+  // override data to populate the user-agent client hints.
+  // 2. Otherwise, if override user-agent contains default user-agent, we
+  // use system default user-agent metadata to populate the user-agent
+  // client hints.
+  // 3. Finally, if the above two cases don't match, we only populate system
+  // default low-entropy client hints.
+  if (base::FeatureList::IsEnabled(blink::features::kUserAgentClientHint)) {
+    // TODO(jfernandez): Implement the user-agent client hints logic
+  }
+
+  // Set overridden user-agent and default client hints metadata if applied.
+  web_contents()->SetUserAgentOverride(override_ua_with_metadata, true);
+
+  content::NavigationController& controller = web_contents()->GetController();
+  for (int i = 0; i < controller.GetEntryCount(); ++i)
+    controller.GetEntryAtIndex(i)->SetIsOverridingUserAgent(true);
+}
```

Comment:
Not sure I get this. Are we backpropagating the UA override change to previous navigation entries? That does not seem correct....